### ISSUE TITLE
Unable to cancel ActionChain action from UI

### ISSUE THREAD

**Issue by codyaray** [StackStorm/st2#2582]
If I setup a simple ActionChain workflow with a long-running action in it, I'm unable to cancel the workflow or the individual long-running action from the UI.

I'm running `st2 ('1.3.2.16',)` installed using the debian packages on Tuesday. Here's a simple definition with which I've been able to reproduce:

```
---
name: "cancel_test"
description: "Test cancel long-running action-chain action from UI and CLI"
enabled: true

runner_type: "action-chain"
entry_point: "workflows/cancel_test.yaml"
```

```
---
chain:
  -
    name: "short_first"
    ref: "core.local"
    parameters:
      cmd: "echo vroom vroom"
    on-success: "slow_second"
  -
    name: "slow_second"
    ref: "core.local"
    parameters:
      cmd: "sleep 90 && echo done"
```

From the UI, when I click "Cancel" on the workflow or expand the workflow and click "Cancel" on the currently running task, nothing appears to happen. Neither the status nor button changes to show cancelled as it normally does.

Querying from the CLI also shows that the workflow and action are still running.

```
cray@svlchi6pstack2:~$ st2 execution list | tail -2
| + 56ead3c8dac007616649e8b7 | default.cancel_test             | st2admin          | running (19s elapsed) | Thu, 17 Mar 2016 15:56:56 UTC |                               |
+----------------------------+-------------------------------+-------------------+-----------------------+-------------------------------+-------------------------------+
cray@svlchi6pstack2:~$ st2 execution get 56ead3c8dac007616649e8b7
id: 56ead3c8dac007616649e8b7
action.ref: default.cancel_test
parameters: None
status: running (26s elapsed)
start_timestamp: 2016-03-17T15:56:56.827066Z
end_timestamp: None
+--------------------------+-----------+-------------+------------+-------------------------------+
| id                       | status    | task        | action     | start_timestamp               |
+--------------------------+-----------+-------------+------------+-------------------------------+
| 56ead3c9dac0073b4af2f47b | succeeded | short_first | core.local | Thu, 17 Mar 2016 15:56:57 UTC |
| 56ead3cadac0073b4af2f47d | running   | slow_second | core.local | Thu, 17 Mar 2016 15:56:58 UTC |
+--------------------------+-----------+-------------+------------+-------------------------------+
```

The CLI behaves as I'd expect when canceling the long-running action.

```
cray@svlchi6pstack2:~$ st2 execution list | tail -2
| + 56ead2c1dac007616649e8b4 | default.cancel_test             | st2admin          | running (5s elapsed) | Thu, 17 Mar 2016 15:52:33 UTC |                               |
+----------------------------+-------------------------------+-------------------+----------------------+-------------------------------+-------------------------------+
cray@svlchi6pstack2:~$ st2 execution get 56ead2c1dac007616649e8b4
id: 56ead2c1dac007616649e8b4
action.ref: default.cancel_test
parameters: None
status: running (12s elapsed)
start_timestamp: 2016-03-17T15:52:33.706605Z
end_timestamp: None
+--------------------------+-----------+-------------+------------+-------------------------------+
| id                       | status    | task        | action     | start_timestamp               |
+--------------------------+-----------+-------------+------------+-------------------------------+
| 56ead2c1dac0073b47961f9c | succeeded | short_first | core.local | Thu, 17 Mar 2016 15:52:33 UTC |
| 56ead2c2dac0073b47961f9e | running   | slow_second | core.local | Thu, 17 Mar 2016 15:52:34 UTC |
+--------------------------+-----------+-------------+------------+-------------------------------+
cray@svlchi6pstack2:~$ st2 execution get 56ead2c2dac0073b47961f9e
id: 56ead2c2dac0073b47961f9e
status: running (15s elapsed)
parameters:
  cmd: sleep 90 && echo done
result: None
cray@svlchi6pstack2:~$ st2 execution cancel 56ead2c2dac0073b47961f9e
action execution with id 56ead2c2dac0073b47961f9e canceled.
cray@svlchi6pstack2:~$ st2 execution get 56ead2c2dac0073b47961f9e
id: 56ead2c2dac0073b47961f9e
status: canceled
parameters:
  cmd: sleep 90 && echo done
result:
  message: Action canceled by user.
  user: st2admin
cray@svlchi6pstack2:~$ st2 execution get 56ead2c1dac007616649e8b4
id: 56ead2c1dac007616649e8b4
action.ref: default.cancel_test
parameters: None
status: canceled
start_timestamp: 2016-03-17T15:52:33.706605Z
end_timestamp: 2016-03-17T15:52:52.114262Z
+--------------------------+-----------+-------------+------------+-------------------------------+
| id                       | status    | task        | action     | start_timestamp               |
+--------------------------+-----------+-------------+------------+-------------------------------+
| 56ead2c1dac0073b47961f9c | succeeded | short_first | core.local | Thu, 17 Mar 2016 15:52:33 UTC |
| 56ead2c2dac0073b47961f9e | canceled  | slow_second | core.local | Thu, 17 Mar 2016 15:52:34 UTC |
+--------------------------+-----------+-------------+------------+-------------------------------+
cray@svlchi6pstack2:~$ st2 execution list | tail -2
| + 56ead2c1dac007616649e8b4 | default.cancel_test             | st2admin          | canceled  | Thu, 17 Mar 2016 15:52:33 UTC | Thu, 17 Mar 2016 15:52:52 UTC |
+----------------------------+-------------------------------+-------------------+-----------+-------------------------------+-------------------------------+
```

Canceling the top-level workflow from the CLI also doesn't behave as I'd expect. But this could be a misunderstanding on my part. I would expect it to cancel the running task and not start any new tasks. But it might let the current task run to completion and just not start the next action? That would also be fine if I'm just assuming incorrectly on this part.

**Comment by Kami** [comment]
Thanks for the report.

As far as canceling a top-level execution of an action chain workflow goes - this will cancel the workflow, but it won't kill the currently running task (we don't have a good and consistent mechanism for this right now). It will wait for the currently running task to complete, but it won't proceed with the next one.

**Comment by Kami** [comment]
As far as canceling a workflow in WebUI goes, this seems like a bug in st2web, but I would need to dig in and try to reproduce it.

I believe we use the same even stream for cancel event notifications so not exactly sure what could go wrong without digging in. I believe that at the very least, the status of the top-level execution should change to canceled, but as mentioned above, the currently running task will still continue to run until it finishes our times out.

/cc @enykeev

**Comment by enykeev** [comment]
@codyaray I'm sorry you're having trouble with st2web.

Kami's correct, it is indeed a bug in st2web. We've shipped 1.3 package with an old API client that doesn't support execution cancelling. The problem has been fixed in https://github.com/StackStorm/st2web/pull/293 and should appear in our package repos as `st2web_1.3.2-4` in mere minutes.
